source/common/http/filter_manager.h

```diff
@@ -966,6 +1002,9 @@ class FilterManager : public ScopeTrackedObject,
   // Indicates which filter to start the iteration with.
   enum class FilterIterationStartState { AlwaysStartFromNext, CanStartFromCurrent };
 
+  absl::optional<bool> createUpgradeFilterChain(const FilterChainFactory& filter_chain_factory,
```

Comment:
I'm not a huge fan of `optional<bool>` because if you have `optional<bool> foo` and do `if (foo)` it's not always obvious to readers if that is testing if `foo` has a value, or if `foo`'s value is `true`. `if (foo)` is different from `if (foo == true)` which is a bit mind bending. We could perhaps define a tri-state enum for this purpose which would be a bit more clear.

Reply:
SGTM.

Reply:
This point is pretty reasonable that I never thought before.